### ISSUE TITLE
Crash if multiple files have errors. 

### ISSUE THREAD

**Issue by Gregroam** [expressjs/multer#335]
For example, if you try to upload 2 files that are both too big, you get a weird uncaught  error
`TypeError: path must be a string
    at TypeError (native)
    at Object.fs.unlink (fs.js:927:11)`

And this seems to be because on the first error, it unlinks the uploaded files, and deletes their filepath and other properties, then on the 2nd error, the filepath property does not exist on the uploaded files, and so this weird uncaught error occurs and crashes the process.

**Comment by RafLeszczynski** [comment]
have the same issue

**Comment by Roam-Cooper** [comment]
Can @LinusU ping us back as to whether they know this issue has been resolved or not, otherwise we can fix via a pull request as we are using this module for a project that will go to production in a few months.

**Comment by LinusU** [comment]
I haven't been able to reproduce this myself so I don't know if it's present or not. If someone could submit a test case that fails I would be more than happy to fix it :)

**Comment by Gregroam** [comment]
Node app
```
var express = require("express");
var app = express();

var multer = require("multer");
var upload = multer({
	dest: __dirname+"/upload",
	limits: {fileSize: 10000}
});

app.post("/upload", function(req, res, next) {
	return upload.array("file")(req,res, function(err){
		if (err) {
			console.log("error",err);
			return res.end("Failed to upload");
		}
		console.log(req.files);
		return res.end("Uploaded sucessfully");
	});
})

app.listen(8080, function() {
	console.log("Listening");
})
```

[Postman Screenshot of request body with 1 file](http://i.imgur.com/HATMi0z.png)
When I try to upload only one file greater than 10,000 bytes. Then I get "Failed to upload" and the console logs as expected:
 `error { [Error: File too large] code: 'LIMIT_FILE_SIZE', field: 'file', storageErrors: [] }`


[Postman Screenshot of request body with 2 files](http://i.imgur.com/D5Npu1t.png)
When I try to upload two files greater than 10,000 bytes each. Then the process crashes with this error.
```
fs.js:927
  binding.unlink(pathModule._makeLong(path), req);
          ^

TypeError: path must be a string
    at TypeError (native)
    at Object.fs.unlink (fs.js:927:11)
    at DiskStorage._removeFile (/Test/node_modules/multer/storage/disk.js:61:6)
    at remove (/Test/node_modules/multer/lib/make-middleware.js:64:19)
    at handleFile (/Test/node_modules/multer/lib/remove-uploaded-files.js:10:5)
    at removeUploadedFiles (/Test/node_modules/multer/lib/remove-uploaded-files.js:25:3)
    at EventEmitter.<anonymous> (/Test/node_modules/multer/lib/make-middleware.js:67:9)
    at EventEmitter.g (events.js:260:16)
    at emitNone (events.js:72:20)
    at EventEmitter.emit (events.js:166:7)
```

**Comment by RafLeszczynski** [comment]
@Roam-Cooper I have the same case so if you already has a fix please share the PR so we won't duplicate work :-)

**Comment by Gregroam** [comment]
@RafLeszczynski I just accidentally discovered a work around. Instead of letting multer handle writing the files to disk, omit the "dest" option so that it keeps the uploaded files in buffer, and then manually write the buffers to files yourself. If there are multiple errors, the process won't crash as it it doesn't try to remove the files multiple times as they are being kept in memory instead.

Example script
```
var express = require("express");
var app = express();
var fs = require("fs");

var multer = require("multer");

//Do not pass "dest" in the options. This will keep the uploaded files in memory instead.
var upload = multer({
	limits: {fileSize: 10000}
});

app.post("/upload", function(req, res, next) {
	return upload.array("file")(req,res, function(err){
		if (err) {
			console.log("error",err);
			return res.end("Failed to upload");
		}
		req.files.forEach(function(file, index) {
			//Generate filepath + filename here however you want
			var filepath = "./upload/file"+index;
			fs.writeFile(filepath, file.buffer);
		});
		return res.end("Uploaded sucessfully");
	});
})

app.listen(8080, function() {
	console.log("Listening");
})
```

**Comment by LinusU** [comment]
Should be fixed #380, released as 1.2.1
